firebase/transformData.js

```diff
@@ -0,0 +1,38 @@
+const getDateFormat = (date) => {
+  const stringDate = `${date.getDate()}. ${
+    date.getMonth() + 1
+  }. ${date.getFullYear()}`;
+
+  return stringDate;
+};
+
+const dateOfPublicationFormat = (data) => {
+  data.forEach((item) => {
+    if (item.dateOfPublication) {
+      item.dateOfPublication = getDateFormat(
+        new Date(item.dateOfPublication.toDate()),
+      );
+    }
+  });
+};
+
+export const transformSnapshot = (snapshot) => {
+  const transformedData = snapshot.docs.map((doc) => ({
+    ...doc.data(),
+    id: doc.id,
+  }));
+  dateOfPublicationFormat(transformedData);
+
+  return transformedData;
+};
+
+export const groupByFirstChar = (objectArray, property) =>
+  objectArray.reduce((acc, obj) => {
+    const key = obj[property].charAt(0);
+    if (!acc[key]) {
+      acc[key] = [];
+    }
+    acc[key].push(obj);
+
+    return acc;
+  }, {});
```

Comment:
We could make the naming a bit clearer. It is a matter of preference, but I am confused about how it actually works, so it is a good rule of thumb :D.

```suggestion
export const groupByFirstChar = (array, groupProperty) =>
  array.reduce((acc, item) => {
    const firstChar = item[groupProperty].charAt(0);
    
    if (!acc[firstChar]) {
      acc[firstChar] = []; // mutation 
    }
    
    acc[firstChar].push(item); // even this mutates

    return acc;
  }, {});
```

Nice function btw, not many people know how to reduce works. Btw, here the mutation of acc is perfectly fine. It is local and actually helps a bit with performance. I can explain further on a call.

Reply:
wait a minute?! I can see some comments inside! :D

Reply:
That was for you! :D Those of course must be obliterated.